### PR TITLE
 build.py : Add "doc/licenses/manifest.json" to packages

### PR DESCRIPTION
**Pull request by johnhaddon**
This will be used to replace the information currently hardcoded in `Gaffer.About`.

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -297,16 +303,29 @@ def walk( project, configs, buildDir ) :
 
 		for pattern in configs[project].get( "manifest", [] ) :
 			for f in glob.glob( os.path.join( buildDir, pattern ) ) :
-				manifest.add( os.path.relpath( f, buildDir ) )
+				files.add( os.path.relpath( f, buildDir ) )
+
+		m = {
+			"name" : project,
+			"url" : configs[project]["url"],
+			"license" : "./" + project,
```
Comment (themissingcow) [inline build.py:311]: We treat `license` as being optional elsewhere, should we be applying the same check here as we do for `credit`?https://github.com/GafferHQ/dependencies/blob/c521ac7f744b7644ce2c6dd91b5adf67e5661470/build.py#L215

**Comment by johnhaddon** [comment]
> Feel free to ignore this, but I think my only niggling feeling is that from a maintenance perspective by muppets like me, build.py is becoming more and more intricate, and relies on 'matching logic' in different functions

Aside from the outrageous claim to be a small comedic animal, that's a reasonable comment. But it's not particularly valuable unless accompanied by specific suggestions for improvement. I think we've debated privately the degree to which we want to "engineer" `build.py`, and I'm still firmly of the opinion that it should have minimal pretensions. I don't want to add additional layers of abstraction unless absolutely necessary.

> L311 relies on the behaviour of L220/L222

I can't come up with anything to help here. We could farm it out to a `licenseLocation()` function, but each client would want different formatting - one wants relative and one wants absolute and it's such a trivial concatenation anyway that I don't much benefit in abstraction.

 > it's not immediately obvious that L293 and L323 need to be kept in sync

Yeah, this should probably have been a variable. But happily, it turns out that we don't need to explicitly include the manifest file in the list anyway, because it's included by virtue of the `doc/licenses` directory being included. So I've removed it completely in aa8c696.

File: build.py (hunk 1 above)
Comment (johnhaddon) [reply build.py]: Indeed - fixed in aa8c696. Really we should enforce that all projects must have a license and then we can simplify this out. The culprit is GafferHQ/resources, which contains demo assets built by IE and a cow of mysterious origins :)

**Comment by johnhaddon** [comment]
I should also document the new "url" and "credits" config options, but that documentation appears only on master I think. I can take care of it when merging this forwards.

**Comment by themissingcow** [comment]
> Aside from the outrageous claim to be a small comedic animal, that's a reasonable comment. But it's not particularly valuable unless accompanied by specific suggestions for improvement. I think we've debated privately the degree to which we want to "engineer" `build.py`, and I'm still firmly of the opinion that it should have minimal pretensions. I don't want to add additional layers of abstraction unless absolutely necessary.

Hehe, true, I guess I wasn't wanting to prescribe the _amount_ of engineering until we decided if we did actually want any more :-)

> I can't come up with anything to help here. We could farm it out to a `licenseLocation()` function, but each client would want different formatting - one wants relative and one wants absolute and it's such a trivial concatenation anyway that I don't see much benefit in abstraction.

This was where my thoughts ended up too... I think now the other changes have removed a bunch of the things that were niggling me, then it's a non issue now anyway.

Thanks for indulging the thought experiment though...

**Pull request merged**